core/src/test/scala/org/apache/spark/deploy/history/ApplicationCacheSuite.scala

```diff
@@ -380,4 +382,27 @@ class ApplicationCacheSuite extends SparkFunSuite with Logging with MockitoSugar
     verify(resp).sendRedirect("http://localhost:18080/history/local-123/jobs/job/?id=2")
   }
 
+  test("redirect respects X-Forwarded-Proto header") {
```

Comment:
Thank you for adding this. It would be better if you follow the guideline (https://spark.apache.org/contributing.html)
```scala
- test("redirect respects X-Forwarded-Proto header") {
+ test("SPARK-27996: redirect respects X-Forwarded-Proto header") {
```

Reply:
I have changed the unit test name